tests/relay/test_relay_dispatcher.py

```diff
@@ -144,3 +159,56 @@ async def test_event_3(self, event_builder):
             ]
         )
         await dispatcher.handle(client_session, event)
+
+    @pytest.mark.asyncio
+    async def test_legal_timestamp(self, event_builder):
+        nip_22_config = nips_config.nip_22
+        if not nip_22_config:
+            return
+        current_time = time.time()
+        legal_timeframe_size = nip_22_config[1] - nip_22_config[0]
+        lower_bound_timestamp = current_time + nip_22_config[0]
+        amount = 5
+        # do not check lower bound as it can fail due to runtime delay
+        legal_timestamps = [lower_bound_timestamp + legal_timeframe_size / amount * i for i in range(1, amount + 1)]
+        await _attempt_timestamp(event_builder=event_builder, timestamps=legal_timestamps, should_save=True)
+
+    @pytest.mark.asyncio
+    async def test_illegal_timestamp(self, event_builder):
+        nip_22_config = nips_config.nip_22
+        if not nip_22_config:
+            return
+        current_time = time.time()
+        illegal_timestamps = [
+            int(current_time + nip_22_config[0] - 60 * 60 * 24 * 365),  # a year before lower limit
+            int(current_time + nip_22_config[0] - 60),  # a minute before lower limit
+            int(current_time + nip_22_config[1] + 60),  # a minute after upper limit
+            int(current_time + nip_22_config[1] + 60 * 60 * 24 * 365)  # a year after upper limit
+        ]
+        await _attempt_timestamp(event_builder=event_builder, timestamps=illegal_timestamps, should_save=False)
+
+    def test_timestamp_validation(self):
+        good_timestamps_params = [
+            (1500, 2000, (-600, 0)),
+            (1700, 2000, (-600, -200)),
+            (190000, 100000, (80000, 100000)),
+            (250000, 100000, (-600, 500000)),
+            (100000, 100000, (0, 1)),
+            (101, 100, (-600, 500000)),
+            (8000, 100000, None),
+        ]
+        bad_timestamps_params = [
+            (1300, 2000, (-600, 0)),
+            (1900, 2000, (-600, -200)),
+            (150000, 100000, (80000, 100000)),
+            (9999999, 100000, (-600, 500000)),
+            (100001, 100000, (0, 1)),
+            (151, 100, (-600, 50)),
+        ]
+        for param_list, expected in [
+            (good_timestamps_params, True),
+            (bad_timestamps_params, False)
+        ]:
+            for param_instance in param_list:
+                result = is_creation_time_valid(*param_instance)
+                assert result == expected
```

Comment:
first - you probably want to use parameterize here :) 
second - its a bit hard to read like that, anyway to improve readability?